examples/kafka/dev-values-tls-all-components-and-kafka-and-oauth2-low-resource.yaml

```diff
@@ -0,0 +1,322 @@
+  # Note: The configured certificate must allow for both server and client use since it is used
+  #       for mTLS. This should be in certificate:
+  #
+  # X509v3 Extended Key Usage:
+  #               TLS Web Server Authentication, TLS Web Client Authentication
+  # If using cert-manager, make sure your certificate includes:
+  #
+  zookeeper:
+    enabled: true
+    createCertificates: true
+    enableHostnameVerification: true
+    tlsSecretName: "pulsar-zookeeper-tls"
+    configureKeystoreWithPasswordFile: true
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  bookkeeper:
+    enabled: true
+    createCertificates: true
+    tlsSecretName: "pulsar-bookkeeper-tls"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  function:
+    enableTlsWithBroker: true
+    createCertificates: true
+    tlsSecretName: "pulsar-function-tls"
+    enableHostnameVerification: true
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  websocket:
+    enableTlsWithBroker: true
+    enableHostnameVerification: true
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  proxy:
+    enableTlsWithBroker: true
+    enableHostnameVerification: false
+    createCertificates: true
+    tlsSecretName: "pulsar-proxy-tls"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  broker:
+    createCertificates: true
+    tlsSecretName: "pulsar-broker-tls"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  # Certificate used for TLS authentication with bookkeeper and zookeeper
+  autoRecovery:
+    enableHostnameVerification: true
+    createCertificates: true
+    tlsSecretName: "pulsar-autorecovery-tls"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  pulsarAdminConsole:
+    enableTlsWithBroker: true
+    createCertificates: true
+    tlsSecretName: "pulsar-adminconsole-tls"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+  ssCaCert:
+    tlsSecretName: "pulsar-ss-ca"
+    certSpec:
+      privateKey:
+        algorithm: "ECDSA"
+        size: 256
+
+# The current function worker code only uses TLS connections to brokers when authentication is enabled, so enable it here.
+enableTokenAuth: true
+restartOnConfigMapChange:
+  enabled: true
+extra:
+  broker: false
+  brokerSts: true
+  proxy: true
+  function: true
+  burnellLogCollector: false
+  # Note that these deployments currently use the plaintext broker ports
+  burnell: true
+  pulsarHeartbeat: true
+  pulsarAdminConsole: true
+
+cert-manager:
+  enabled: true
+
+createCertificates:
+  selfSignedPerComponent:
+    enabled: true
+
+openid:
+  enabled: true
+  # From token generated in Okta UI or other method:
+  issuerUrl: https://dev-42506116.okta.com/oauth2/aus7ypk6sjvgF4l615d7
+  scope: pulsar_client_m2m
+  client_id: "0oa7ypwvxnvo9xnDd5d7"
```

Comment:
Reminder to replace secrets with placeholders and write in the instructions where to place the secrets and the URL for Okta. In the instructions and demo script, we can do something like helm install --set path.to.secret=SECRET-VALUE …..